(docs)(MODULES-4752)(MODULES-4220) Update README with current puppet type

glennsarti opened the pull request:
Previously bindings were added to the iis_site type and provider however there
was no documentation on the README.  This commit:
- Adds documentation on how to use the bindings section in iis_site, including
  examples
- Uses alphabetical listing for the list of classes
- Fixes a typo for the title of the iis_feature resource documentation

Previously in commit 76fca17 iis_application_pool was updated with many
additional properties.  This commit updates the README file to reflect all
of the additions made.

glennsarti committed:
(MODULES-4752) Update documentation with bindings

Previously bindings were added to the iis_site type and provider however there
was no documentation on the README.  This commit:
- Adds documentation on how to use the bindings section in iis_site, including
  examples
- Uses alphabetical listing for the list of classes
- Fixes a typo for the title of the iis_feature resource documentation

glennsarti committed:
(MODULES-4220) Document additional properties for iis_application_pool

Previously in commit 76fca174d9d iis_application_pool was updated with many
additional properties.  This commit updates the README file to reflect all of
the additions made.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
 
+#### `auto_start`
+
+When true, indicates to the World Wide Web Publishing Service (W3SVC) that the application pool should be automatically started when it is created or when IIS is started. Valid options `true` or `false`
```
values `true` and `false` should be in backticks throughout

Needs a period at the end of the line; this is true throughout the README

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
 #### `managed_pipeline_mode`
 
-The managedpipelinemode of the ApplicationPool. Must be either `Integrated` or `Classic`
+Specifies the request-processing mode that is used to process requests for managed content. Must be either `Integrated` or `Classic`
```
I think Integrated and Classic might actually be string values? If so, they should be in single quotes instead of backticks.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `managed_runtime_loader`
+
+Specifies the managed loader to use for pre-loading the the application pool.Note: This property was added in IIS 7.5.The default value is `webengine4.dll`
```
String values should be in single quotes; needs a period at the end. Needs space between sentences.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `pass_anonymous_token`
+
+When true, the Windows Process Activation Service (WAS) creates and passes a token for the built-in IUSR anonymous user account to the Anonymous authentication module. The Anonymous authentication module uses the token to impersonate the built-in account. When false, the token will not be passed.Note: The IUSR anonymous user account replaces the IIS_MachineName anonymous account. The IUSR account ...
```
true and false should be in backticks; change "will not be" to "is not", needs a space between "passed." and "Note:", and needs a period at the end. OR just replace with:

When `true`, the Windows Process Activation Service (WAS) creates and passes a token for the built-in IUSR anonymous user account to the Anonymous authentication module. The Anonymous authentication module uses the token to impersonate the built-in account. When `false`, the token is not passed. Note: The IUSR anonymous user account replaces the IIS_MachineName anonymous account. The IUSR account can be used by IIS or other applications. It does not have any privileges assigned to it during setup. Valid options `true` or `false`.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `start_mode`
+
+Specifies the startup type for the application pool. Valid options `:OnDemand` or `:AlwaysRunning`
```
If these valid options are passed as strings (which I think they are), they should be in single quotes.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `cpu_action`
+
+Configures the action that IIS takes when a worker process exceeds its configured CPU limit. Valid options `:NoAction`, `:KillW3wp`, `:Throttle` or `:ThrottleUnderLoad`.
```
Again, if the values are passed as strings, should be in single quotes rather than backticks.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+Important: The `cpu_reset_interval` value must be greater than the time between logging operations, otherwise IIS will reset counters before logging has occurred, and process accounting will not occur.
+
+Note: Because process accounting in IIS uses Windows job objects to monitor CPU times for the whole process, process accounting will only log and throttle applications that are isolated in a separate process from IIS
```
"...process accounting logs and throttles only applications that are..."

glennsarti replied:
Unless I'm missing something, I think your change doesn't make sense.  It could be case that that sentence should be fragmented.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+Specifies the hexadecimal processor mask for multi-processor computers, which indicates to which CPU the worker processes in an application pool should be bound. Before this property takes effect, the `cpu_smp_affinitized` property must be set to true for the application pool.
+
+Note: On 64-bit computers, the cpu_smp_processor_affinity_mask property contains the low-order DWORD for the processor mask, and the `cpu_smp_processor_affinity_mask2` property contains the high-order DWORD for the processor mask. On 32-bit computers, the `cpu_smp_processor_affinity_mask2` property has no effect.
```
`cpu_smp_processor_affinity_mask`

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+If you set the value to 1 (which corresponds to 00000000000000001 in binary), the worker processes in an application pool run on only the first processor. If you set the value to 2 (which corresponds to 0000000000000010 in binary), the worker processes run on only the second processor. If you set the value to 3 (which corresponds to 0000000000000011 in binary) the worker processes run on both the ...
+
+Note: Do not set this property to 0. Doing so disables symmetric multiprocessing (SMP) affinity and creates an error condition. This means that processes running on one CPU will not remain affiliated with that CPU throughout their lifetime
```
needs a period at the end.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+Note: Do not set this property to 0. Doing so disables symmetric multiprocessing (SMP) affinity and creates an error condition. This means that processes running on one CPU will not remain affiliated with that CPU throughout their lifetime
+
+#### `cpu_smp_processor_affinity_mask2`
```
needs a line space after this header

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `identity_type`
+
+Specifies the account identity under which the application pool runs. Note: Starting in IIS 7.5 the default value is `:ApplicationPoolIdentity`. (In IIS 7.0 the default value was `:NetworkService`).  Valid options `:ApplicationPoolIdentity`, `:LocalService`, `:LocalSystem`, `:NetworkService`, or `:SpecificUser`
```
These values are likely passed as strings, so should be in single quotes rather than backticks.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `idle_timeout`
+
+Specifies how long (in minutes) a worker process should run idle if no new requests are received and the worker process is not processing requests. After the allocated time passes, the worker process should request that it be shut down by the WWW service
```
change "...the worker process should request..." to "...the worker process requests..."

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `idle_timeout_action`
+
+Specifies the action to perform when the `idle_timeout` duration has been reached. Before IIS 8.5, a worker process that was idle for the duration of the `idle_timeout` property would be terminated. After IIS 8.5, you have the choice of terminating a worker process that reaches the `idle_timeout` limit, or suspending it by moving it from memory to disk. Suspending a process will likely take less  ...
```
valid options, if passed as strings, should be in single quotes

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+Specifies which action taken in the process gets logged to the Event Viewer. In IIS 8.0, the only action that applies is the idle timeout action, in which the process is terminated because it was idle for the idleTimeout period
+
+#### `logon_type`
+Specifies the logon type for the process identity. (For additional information about logon types, see the LogonUser Function topic on Microsoft's MSDN Web site). Valid options `:LogonBatch` or `:LogonService`
```
valid options, if passed as strings, should be in single quotes

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+Indicates the maximum number of worker processes that would be used for the application pool.
+
+A value of '1' indicates a maximum of a single worker process for the application pool. This would be the setting on a server that does not have NUMA nodes.
```
If the values in this list are integers, they should not have single quotes or backticks

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `set_profile_environment`
+
+When setProfileEnvironment is set to True, WAS creates an environment block to pass to CreateProcessAsUser when creating a worker process. This ensures that the environment is set based on the user profile for the new process. Valid options `true` or `false`
```
`true` rather than True

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `password`
+
+Specifies the password associated with the `user_name` property. This property is only necessary when the value of `identity_type` is `:SpecificUser`.  Note: To avoid storing unencrypted password strings in configuration files, this uses AppCmd.exe. This encrypts the password automatically before it is written to the XML configuration files. This provides better password security than storing une ...
```
if `:SpecificUser` is a string value, it should be in single quotes instead of backticks

Also, this sentence has a rogue quotation mark at the end.

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `load_balancer_capabilities` do
+
+Specifies behavior when a worker process cannot be started, such as when the request queue is full or an application pool is in rapid-fail protection. Valid options `HttpLevel` or `TcpLevel`
```
String values for valid options in single quotes

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### rapid_fail_protection
+
+Setting to true instructs the WWW service to remove from service all applications that are in an application pool when:
```
`true`

jbondpdx commented on README.md:
```diff
@@ -89,17 +89,255 @@ Name of the application pool.
+
+#### `log_event_on_recycle`
+
+Specifies that IIS should log an event when an application pool is recycled. The `log_event_on_recycle` property must have a bit set corresponding to the reason for the recycle if IIS is to log the event. The logEventOnRecycle property can have one or more of the following possible values: `ConfigChange`, `IsapiUnhealthy`, `Memory`, `OnDemand`, `PrivateMemory`, `Requests`, `Schedule` and `Time`.  ...
```
single quotes for values

jbondpdx commented on README.md:
```diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+
+###### Examples:
+
+Use the default HTTP port
```
needs a space between the line and the code sample

jbondpdx commented on README.md:
````diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+}
+```
+
+Use the default HTTPS port
````
needs a space between the line and the code sample

jbondpdx commented on README.md:
````diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+}
+```
+
+Multiple bindings, one for HTTP and another for HTTPS on non-default port
````
needs a space between the line and the code sample

jbondpdx commented on README.md:
````diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+}
+```
+
+Each binding is a hash with the following keys
````
should have a colon after "keys"

jbondpdx commented on README.md:
```diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+
+**Only valid with a protocol of https**
+
+Specifies the thumbprint, also known as the certificatehash, of the certificate that will be used by the IIS site. You can retrieve the thumbprint for a certificate using PowerShell, for example:
```
delete "that will be"

jbondpdx commented on README.md:
```diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+
+**Only valid with a protocol of https**
+
+Specifies the certificate store to search for the relevant `certificatehash`.  Typically this `MY` which is the Personal certificate store for the Local Machine.  You can retrieve the list of stores using PowerShell:
```
I think there is an "is" missing. (Typically, this is `MY`...)

jbondpdx commented on README.md:
````diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+...
+```
+
+###### `protocol`
````
needs a space after the header

jbondpdx commented on README.md:
```diff
@@ -126,14 +364,135 @@ The name of an ApplicationPool for this IIS Web Site
+
+**Only valid with a protocol of https**
+
+The sslflags parameter accepts integer values from 0 to 3.
```
`sslflags` parameter should be in backticks, but the integer values in the following bullet list don't need backticks (or single quotes); just leave them unmarked.

jbondpdx reviewed (changes requested):
A few specific editorial and formatting suggestions, but mostly comments to address, but mostly there are two things throughout:

* Sentences throughout need periods
* Values that are strings should have single quotes, rather than backticks (afaik, only true, false, and undef normally get backticks)

glennsarti committed:
(doc) Update README with markdown style changes

This commit modifies the README file to fix minor markdown style violations and
spelling errors.

glennsarti commented:
@jbondpdx Modifications made as per your review.  Also added in some other markdown style changes too that we both missed.

jbondpdx reviewed (approved):
lgtm!
